### ISSUE TITLE
RDS: CfnDbProxyEndpoint.targetRole has disappeared in 2.64.0

### ISSUE THREAD

**Issue by apancutt** [aws/aws-cdk#24092]
### Describe the bug

The `targetRole` property has unexpectedly been removed from `CfnDbProxyEndpoint` in 2.64.0 (exists in 2.63.0).

### Expected Behavior

Prop should exist in `rds.generated.d.ts`

### Current Behavior

Prop does not exist in `rds.generated.d.ts`

### Reproduction Steps

Attempt to create a read-only proxy endpoint...

```ts
new aws_rds.CfnDBProxyEndpoint(scope, 'ReadOnlyProxyEndpoint', {
  dbProxyEndpointName: 'foo-endpoint-ro',
  dbProxyName: 'foo',
  targetRole: 'READ_ONLY',  // 💥 bang
  vpcSecurityGroupIds: [],
  vpcSubnetIds: [],
});
```

### Additional Information/Context

The `TargetRole` property has also disappeared from the [CFN docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbproxyendpoint.html), but there is no reference to this change in the [Release History](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/ReleaseHistory.html).

The `--target-role` option is still supported by [the CLI](https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-proxy-endpoint.html).

`TargetRole` still exists in [`cfnspec`](https://raw.githubusercontent.com/aws/aws-cdk/main/packages/%40aws-cdk/cfnspec/spec-source/cfn-docs/cfn-docs.json).

[This guide](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/rds-proxy-endpoints.html#rds-proxy-endpoints.CreatingEndpoint) still exists, which describes using the Target Role to force an endpoint to be read-only.

### CDK CLI Version

2.64.0

### Node.js Version

19.6.0

### OS

macOS

### Language

Typescript

### Language Version

4.9.5

**Comment by peterwoodworth** [comment]
Thanks for reporting this @apancutt, I can confirm this property has been removed in the most recent version.

This happened in [this PR](https://github.com/aws/aws-cdk/pull/23984/files#diff-448833823999f11d01885d768d952386baf95b8d41a7096616d5d96e81acce0d) I believe: there's a couple other changes in this spec we should look into as well. We'll look into this and get back to you 🙂

**Comment by apancutt** [comment]
Affects 2.65.0 too

**Comment by vumdao** [comment]
Tried to find workaround but it seems that the the attribute is readonly so not able to set `targetRole`

**Comment by yeti-g** [comment]
@peterwoodworth https://github.com/aws-cloudformation/cfn-lint/pull/2581 was merged so can you revert the offending code from the PR that you identified?

**Comment by RomainMuller** [comment]
There are currently other issues preventing the update to the CloudFormation Spec in CDK. We are actively working on addressing those as quickly as possible. I hope we can address it this week, but this is a multi-team coordinated effort and I don't have visibility on all the work streams involved.

**Comment by jmauerhan** [comment]
> There are currently other issues preventing the update to the CloudFormation Spec in CDK. We are actively working on addressing those as quickly as possible. I hope we can address it this week, but this is a multi-team coordinated effort and I don't have visibility on all the work streams involved.

Thanks! I think I might be alright to use 2.63 instead so I'll try that - we're on 2.30 right now so 2.63 is still a good update.

**Comment by bartcallant** [comment]
[v2.70.0](https://github.com/aws/aws-cdk/releases/tag/v2.70.0) was released a couple of hours ago and brought back the property.

**Comment by apancutt** [comment]
Confirmed fixed in 2.70.0 - thanks.
